app/controllers/git_api/concept_exercises_controller.rb

```diff
@@ -10,6 +10,9 @@ def create
         return render_error(:"too_large_#{param}") if params[param].size > 1.megabyte
       end
 
+      # Validate size of optional fields
+      return render_error(:"too_large_introduction_markdown") if params[:introduction_markdown].present? && params[:introduction_markdown].size > 1.megabyte
```

Comment:
```suggestion
      return render_error(:too_large_introduction_markdown) if params[:introduction_markdown].present? && params[:introduction_markdown].size > 1.megabyte
```

Reply:
You could also use either `try` or `&.` with a `to_i` if you preferered. Just makes the line a bit more readable, but I don't mind either way.

```suggestion
      return render_error(:"too_large_introduction_markdown") if params[:introduction_markdown]&.size.to_i > 1.megabyte
```